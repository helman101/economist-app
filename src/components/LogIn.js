import React, { useState } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import 'react-router-dom';
import { toast } from 'react-toastify';
import { userRequest } from '../API/apiRequests';

const LogIn = (props) => {
  const [userEmail, setUserEmail] = useState('');
  const [userPassword, setUserPassword] = useState('');

  const handleChange = (e) => {
    const { name, value } = e.target;

    if (name === 'email') {
      setUserEmail(value);
    } else {
      setUserPassword(value);
    }
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (userEmail === '' || userPassword === '') {
      toast.error(`${userEmail === '' ? 'Email' : 'Password'} can't be blank `, {
        position: 'top-right',
        autoClose: 5000,
        hideProgressBar: false,
        closeOnClick: true,
        pauseOnHover: true,
        draggable: true,
        progress: undefined,
        theme: 'colored',
      });
    } else {
      const params = { email: userEmail, password: userPassword };
      props.dispatch(userRequest(params, props.history.push));
    }
  };

  return (
    <div className="session d-flex justify-content-center align-items-center">
      <div className="form p-2">
        <div className="title mb-4">Log In</div>
        <form className="d-flex flex-column">
          <input type="email" id="email" name="email" placeholder="Email" onChange={handleChange} className="my-1 form-control" />
          <input type="password" id="password" name="password" placeholder="Password" onChange={handleChange} className="my-1 form-control" />
          <a href="/SignUp" className="mt-2 mb-3 text-center link-danger">New here? Join Us!</a>
          <button type="submit" onClick={handleSubmit} className="btn btn-danger text-white fw-bold">SUBMIT</button>
        </form>
      </div>
    </div>
  );
};

LogIn.propTypes = {
  dispatch: PropTypes.func.isRequired,
  history: PropTypes.shape({ push: PropTypes.func.isRequired }).isRequired,
};

export default connect()(LogIn);
