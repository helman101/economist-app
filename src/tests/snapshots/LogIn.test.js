import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { createMemoryHistory } from 'history';
import { MemoryRouter } from 'react-router';
import defaultState from '../../store/defaultStore';
import LogIn from '../../components/LogIn';

const mockStore = configureStore([]);

describe('LogIn', () => {
  let store;
  let component;
  let history;

  beforeEach(() => {
    history = createMemoryHistory();
    store = mockStore(defaultState);
    component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <LogIn history={history} />
        </MemoryRouter>
      </Provider>,
    );
  });

  it('should render propertly', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
