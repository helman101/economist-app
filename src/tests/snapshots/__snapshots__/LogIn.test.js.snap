// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogIn should render propertly 1`] = `
<div
  className="session d-flex justify-content-center align-items-center"
>
  <div
    className="form p-2"
  >
    <div
      className="title mb-4"
    >
      Log In
    </div>
    <form
      className="d-flex flex-column"
    >
      <input
        className="my-1 form-control"
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        type="email"
      />
      <input
        className="my-1 form-control"
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
      />
      <a
        className="mt-2 mb-3 text-center link-danger"
        href="/Signup"
        onClick={[Function]}
      >
        New here? Join Us!
      </a>
      <button
        className="btn btn-danger text-white fw-bold"
        onClick={[Function]}
        type="submit"
      >
        SUBMIT
      </button>
    </form>
  </div>
</div>
`;
